### PR TITLE
Extract closure logic, Document and Add tests to simulation package

### PR DESCRIPTION
**Pull request by giraugh**
You should check out the docs :eyes:

### CODE REVIEW

File: packages/simulation/src/index.ts [hunk 1]
```diff
@@ -1,4 +1,6 @@
-export { default as simulateFSA } from './simulateFSA'
-export { default as expandReadSymbols } from './expandReadSymbols'
-export { default as validTransitions } from './validTransitions'
-export { default as parseRead } from './parseRead'
+export * from './simulateFSA'
+export * from './resolveGraph'
+export * from './validTransitions'
+export * from './parseRead'
+export * from './closureWithPredicate'
+export * from './types.d'
```
Comment (GRA0007) [inline packages/simulation/src/index.ts:6]: Was this a breaking change for the frontend?

**Review by GRA0007 — APPROVED** [review]
Looks good but this might need to be tested with the frontend, see my comment

**Comment by giraugh** [comment]
Tested with frontend, fixed an issue with the Info tables and added a warning about unreachable final states

**Review by GRA0007 — APPROVED** [review]
looks great!

**Pull request merged**